pkgs/development/tools/rust/rust-analyzer/default.nix

```diff
@@ -1,12 +1,18 @@
 { pkgs, callPackage }:
 
 {
-  rust-analyzer-unwrapped = callPackage ./generic.nix rec {
-    rev = "2020-08-24";
+  rust-analyzer-unwrapped = callPackage (import ./generic.nix rec {
+    rev = "2020-09-07";
     version = "unstable-${rev}";
-    sha256 = "11q5shrq55krgpj7rjfqw84131j5g55zyrwww3cxcbr8ndi3xdnf";
-    cargoSha256 = "15kjcgxmigm0lwbp8p0kdxax86ldjqq9q8ysj6khfhqd0173184n";
-  };
+    sha256 = "1yz3s40m4snw3h480cz562mbh5yrsqliynchra0wvyhajg9n8w0s";
+    cargoSha256 = "08b41y8hnvm3yqiq6zs295nr4gx09r6p8fflrfyl35ihv67a7jwn";
+
+    patches = [
+      # FIXME: Temporary fix for our rust 1.45.0 since rust-analyzer requires 1.46.0
+      ./no-loop-in-const-fn.patch
+      ./no-option-zip.patch
+    ];
```

Comment:
I think this would better belong in `generic.nix`, along with other build instructions.

Reply:
:thinking: But it's temporary.

Well, I'm not quite sure what should be put in generic file. You mean all about "how to build", yeah?

Reply:
I'd expect most patches are supposed to be temporary workarounds, so it wouldn't be a problem as long as there's a comment explaining why they're necessary. If you want to make absolutely sure that you remove the patch when it's no longer needed, I guess you could do something like this:

```nix
  patches = with lib;
    if versionAtLeast (getVersion rustPlatform.rust.rustc) "1.46.0" then [
      # FIXME: Temporary fix for our rust 1.45.0 since rust-analyzer requires 1.46.0
      ./no-loop-in-const-fn.patch
      ./no-option-zip.patch
    ] else warn "Some patches for ${pname} needs to be removed." [];
```

This would exclude the patches and warn you about it once `rustc` reaches 1.46.0.

> You mean all about "how to build", yeah?

Yes.

Reply:
> If you want to make absolutely sure that you remove the patch when it's no longer needed, I guess you could do something like this

These patches just do the same thing in a more compatible way and don't change any semantics. I think it's not quite necessary to do that check.

Reply:
That makes sense.